### PR TITLE
Add MEI test cases and plans from checkbox-iiotg-provider and iiotg provider (New)

### PR DESCRIPTION
**Pull request by LiaoU3**
## Description

I move the MEI related test cases from [checkbox-iiotg-provider](https://git.launchpad.net/~checkbox-dev/checkbox-iiotg/+git/checkbox-provider-intliotg/tree/units/mei) to the iiotg provider in checkbox monorepo.

Note: I didn't include the original MEI jobs since I didn't test them in the IIOTG project and I am also not sure if it's suitable nowadays. If you have any concern please let me know.
## Resolved issues
N/A

## Documentation

N/A

## Tests
I run testplan intel-mei-full or you can say Intel Management Engine Interface tests on EHL ([202306-31656](https://certification.canonical.com/hardware/202306-31656/)) which support MEI.

submission: https://certification.canonical.com/hardware/202306-31656/submission/348566/

### CODE REVIEW

File: providers/base/units/mei/test-plan.pxu [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+bootstrap_include:
+
+
+id: intel-mei-automated
+_name: Intel Management Engine Interface tests (automated)
+unit: test plan
+include:
+    intel-mei/check-module
+    intel-mei/check-device
+    intel-mei/get-mei-firmware-version
```
Comment (pieqq) [inline providers/base/units/mei/test-plan.pxu:23]:
```suggestion
    mei/check-module
    mei/check-device
    mei/get-firmware-version
```

File: providers/base/bin/mei.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return data
+
+    def close(self):
+        os.close(self._mei_obj)
+
+
+def get_mei_firmware_version():
+    mei_fw_uuid = "8e6a6715-9abc-4043-88ef-9e39c6f63e0f"
+    mei_fw_ver_req = 0x000002FF
+    mei_fw_ver_rep_length = 28
```
Comment (pieqq) [inline providers/base/bin/mei.py:62]: Not sure how these values are defined, nor where do they come from. If you have some public source, please add a comment above to provide more explanation.

**Comment by pieqq** [comment]
@LiaoU3 I think you messed up something with your last push, because I see tons of unrelated changes. Please rebase your work on `main` and push again. Thanks!

File: providers/base/bin/mei.py (hunk 2 above)
Comment (LiaoU3) [reply providers/base/bin/mei.py]: I've left some comment about these fixed variables.

**Comment by LiaoU3** [comment]
@pieqq Please review this when you have time.

**Review by pieqq — APPROVED** [review]
All good, thanks!

**Pull request merged**